src/iter/par_bridge.rs

```diff
@@ -1,13 +1,18 @@
 use crossbeam_deque::{Steal, Stealer, Worker};
 
 use std::sync::atomic::{AtomicBool, AtomicUsize, Ordering};
-use std::sync::{Mutex, TryLockError};
+use std::sync::{RwLock, TryLockError};
 use std::thread::yield_now;
 
 use crate::current_num_threads;
 use crate::iter::plumbing::{bridge_unindexed, Folder, UnindexedConsumer, UnindexedProducer};
 use crate::iter::ParallelIterator;
 
+/// A wrapper around rwlock where the user basically pinky-promises to never call read()
+struct ReentrantMutex<T>(RwLock<T>);
+unsafe impl<T> Send for ReentrantMutex<T> where T: Send {}
+unsafe impl<T> Sync for ReentrantMutex<T> where T: Send {}
+
```

Comment:
this stuff is copypasted from mutex guarantees. I *think* this is safe, as I don't think reentrancy on a `!Sync` T causes problems as long as you're in the same thread?

Reply:
Yeah I think this is safe: https://amanieu.github.io/parking_lot/parking_lot/struct.ReentrantMutex.html